File: src/main/java/br/edu/ifes/tads/legacy/SaleResult.java
package br.edu.ifes.tads.legacy;

public class SaleResult {
    private final double subtotal;
    private final double discount;
    private final double shipping;
    private final double total;
    private final String receipt;

    public SaleResult(double subtotal, double discount, double shipping, double total, String receipt) {
        this.subtotal = subtotal;
        this.discount = discount;
        this.shipping = shipping;
        this.total = total;
        this.receipt = receipt;
    }

    public double getSubtotal() { return subtotal; }
    public double getDiscount() { return discount; }
    public double getShipping() { return shipping; }
    public double getTotal() { return total; }
    public String getReceipt() { return receipt; }
}

File: src/Legacy/SaleResult.php
<?php

declare(strict_types=1);

namespace CleanCodeLab\Legacy;

final class SaleResult
{
    public function __construct(
        public float $subtotal,
        public float $discount,
        public float $shipping,
        public float $total,
        public string $receipt,
    ) {
    }
}
